features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/lifecycle_api_router.jag

```diff
@@ -21,553 +21,805 @@
-    var shortName = context.params.type;
-    var id = context.params.id;
-    var artifactManager = rxtManager.getArtifactManager(shortName);
-
-    try {
-      var artifact = artifactManager.get(id);
-      artifactManager.checkItem(checkListItemIndex, artifact);
-      var msg = {};
-      msg['status'] = 'ok';
-    } catch (e) {
-      log.debug('The asset with id: ' + id + ' could not be found.The following exception was thrown: ' + e);
-      response.sendError(404, 'The asset with id: ' + id + ' could not be found.Please refer to the server log.');
-    }
-  });
-
-  routeManager.register('DELETE', 'publisher', '/publisher/api/lifecycle/{type}/{id}', function(context) {
-
-    var shortName = context.params.type;
-    var id = context.params.id;
-    var artifactManager = rxtManager.getArtifactManager(shortName);
-    try {
-      var artifact = artifactManager.get(id);
-      artifactManager.detachLifecycle(artifact);
-
-      var msg = {};
-      msg['status'] = 'ok';
-
-      print(msg);
-    } catch (e) {
-      log.debug('The asset with id: ' + id + ' could not be found.The following exception was thrown: ' + e);
-      response.sendError(404, 'The asset with id: ' + id + ' could not be found.Please refer to the server log.');
-    }
-
-  });
+require('/modules/publisher.js').exec(function (ctx) {
+
+	var config = require('/config/publisher.json');
+	var caramel = require('caramel');
+	var router = require('/modules/router-g.js').router();
+	var routeManager = new router.Router();
+	var log = new Log();
+	var permissions = require('/modules/permissions.js').permissions;
+
+	routeManager.register('POST', 'publisher', '/publisher/api/lifecycle/{type}/{id}', function (context) {
+		var shortName = context.params.type;
+		var id = context.params.id;
+		var lifecycle = context.post.lifecycle;
+		//var autoPromote=context.post.promote||false;    //If promote is not specified do not auto promote
+		var invokeAction = '';
+		//Check the config for a lifeCycleBehaviour block
+		utility.isPresent(config, 'lifeCycleBehaviour', function (lifeCycleBehaviour) {
+			utility.isPresent(lifeCycleBehaviour, lifeCycle, function (lifeCycleData) {
+				utility.isPresent(lifeCycleData, 'onAttach', function (onAttach) {
+					invokeAction = onAttach.action || '';
+				});
+			});
+
+		});
+
+		if (!lifecycle) {
+			response.sendError(400, 'Missing required parameter: lifecycle');
+			return;
+		}
+
+		var artifactManager = rxtManager.getArtifactManager(shortName);
+		try {
+			var artifact = artifactManager.get(id);
+			artifactManager.attachLifecycle(lifecycle, artifact);
+			//Perform the action if the user has specified an action
+			if (invokeAction != '') {
+				artifactManager.promoteLifecycleState(invokeAction, artifact);
+			}
+
+			var msg = {};
+			msg['status'] = 'ok';
+
+			print(msg);
+		} catch (e) {
+			log.debug('The asset with id: ' + id + ' could not be found.The following exception was thrown: ' + e);
```

Comment:
We check for isDebugEnabled() before adding debug logs. Better follow same here, in all places that includes debug logs.